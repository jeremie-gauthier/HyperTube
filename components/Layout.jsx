import React from "react";
import styles from "./Layout.module.scss";

export default function Layout({ children }) {
	return <div className={styles.container}>{children}</div>;
}
