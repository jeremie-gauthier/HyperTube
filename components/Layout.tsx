import Navbar from "@/components/Navbar";

type LayoutProps = {
  children: React.ReactNode;
};

export default function Layout({ children }: LayoutProps): JSX.Element {
  return (
    <div className="h-screen" data-testid="layout-div">
      <Navbar />
      {children}
    </div>
  );
}
