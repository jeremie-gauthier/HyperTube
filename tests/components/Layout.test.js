import React from "react";
import Layout from "@/components/Layout";
import renderer from "react-test-renderer";

test("<Layout /> should render correctly", () => {
  const tree = renderer.create(<Layout />).toJSON();
  expect(tree).toMatchSnapshot();
});
