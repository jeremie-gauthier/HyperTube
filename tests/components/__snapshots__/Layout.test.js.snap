// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render correctly 1`] = `
<div
  className="h-screen"
  data-testid="layout-div"
>
  <div
    className="flex flex-row container"
    data-testid="navbar-div"
  >
    <div
      className="flex flex-row content"
      data-testid="flex-row"
    >
      <h1>
        HYPERTUBE
      </h1>
      <a
        data-testid="link-home"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Home
      </a>
      <a
        href="/movies/favorites"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        My list
      </a>
      <div
        className="flex flex-row searchInput"
        data-testid="flex-row"
      >
        <svg
          fill="none"
          height="12"
          viewBox="0 0 11 12"
          width="11"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.848 10.375L8.707 8.038a.494.494 0 0 0-.365-.164h-.35a5.138 5.138 0 0 0 .945-3c0-2.692-2-4.874-4.469-4.874C2 0 0 2.182 0 4.875c0 2.692 2 4.874 4.468 4.874a4.217 4.217 0 0 0 2.75-1.031V9.1c0 .15.054.293.15.398l2.142 2.337c.202.22.529.22.729 0l.607-.663a.602.602 0 0 0 .002-.797zm-6.38-2.5c-1.518 0-2.75-1.341-2.75-3 0-1.657 1.23-3 2.75-3s2.75 1.34 2.75 3c0 1.656-1.229 3-2.75 3z"
            fill="#fff"
          />
        </svg>
        <input
          className="text-white placeholder-white placeholder-bold bg-black"
          placeholder="Search"
        />
      </div>
    </div>
    <div
      className="flex flex-row space-x-4"
      data-testid="flex-row"
    >
      <div
        className="frame"
      >
        <span>
          AD
        </span>
      </div>
      <div
        className="frame"
        data-testid="lang-flag"
      >
        <span>
          lg
        </span>
      </div>
    </div>
  </div>
</div>
`;
